sdk/core/azure-core/test/ut/websocket_test.cpp

```diff
@@ -0,0 +1,686 @@
+    EXPECT_EQ(WebSocketFrameType::BinaryFrameReceived, response->FrameType);
+    EXPECT_THROW(response->AsPeerCloseFrame(), std::logic_error);
+    EXPECT_THROW(response->AsTextFrame(), std::logic_error);
+    auto textResult = response->AsBinaryFrame();
+    EXPECT_EQ(binaryData, textResult->Data);
+
+    // Close the socket gracefully.
+    testSocket.Close();
+  }
+
+  {
+    WebSocket testSocket(Azure::Core::Url("http://localhost:8000/echotest?fragment=true"));
+
+    testSocket.Open();
+
+    std::vector<uint8_t> binaryData{1, 2, 3, 4, 5, 6};
+
+    testSocket.SendFrame(binaryData, true);
+
+    std::vector<uint8_t> responseData;
+    std::shared_ptr<Azure::Core::Http::WebSockets::WebSocketFrame> response;
+    do
+    {
+      response = testSocket.ReceiveFrame();
+      EXPECT_EQ(WebSocketFrameType::BinaryFrameReceived, response->FrameType);
+      auto binaryResult = response->AsBinaryFrame();
+      responseData.insert(responseData.end(), binaryResult->Data.begin(), binaryResult->Data.end());
+    } while (!response->IsFinalFrame);
+
+    auto textResult = response->AsBinaryFrame();
+    EXPECT_EQ(binaryData, responseData);
+
+    // Close the socket gracefully.
+    testSocket.Close();
+  }
+}
+
+template <size_t N> void EchoRandomData(WebSocket& socket)
+{
+  std::vector<uint8_t> sendData = Azure::Core::Http::WebSockets::_detail::GenerateRandomBytes(N);
+
+  socket.SendFrame(sendData, true);
+
+  std::vector<uint8_t> receiveData;
+
+  std::shared_ptr<WebSocketFrame> response;
+  do
+  {
+    response = socket.ReceiveFrame();
+    EXPECT_EQ(WebSocketFrameType::BinaryFrameReceived, response->FrameType);
+    auto binaryResult = response->AsBinaryFrame();
+    receiveData.insert(receiveData.end(), binaryResult->Data.begin(), binaryResult->Data.end());
+  } while (!response->IsFinalFrame);
+
+  // Make sure we get back the data we sent in the echo request.
+  EXPECT_EQ(sendData.size(), receiveData.size());
+  EXPECT_EQ(sendData, receiveData);
+}
+
+TEST_F(WebSocketTests, VariableSizeEcho)
+{
+  {
+    WebSocket testSocket(Azure::Core::Url("http://localhost:8000/echotest"));
+
+    testSocket.Open();
+    {
+      EchoRandomData<100>(testSocket);
+      EchoRandomData<124>(testSocket);
+      EchoRandomData<125>(testSocket);
+      // The websocket protocol treats lengths of 125, 126 and > 127 specially.
+      EchoRandomData<126>(testSocket);
+      EchoRandomData<127>(testSocket);
+      EchoRandomData<128>(testSocket);
+      EchoRandomData<1020>(testSocket);
+      EchoRandomData<1021>(testSocket);
+      EchoRandomData<1022>(testSocket);
+      EchoRandomData<1023>(testSocket);
+      EchoRandomData<1024>(testSocket);
+      EchoRandomData<2048>(testSocket);
+      EchoRandomData<4096>(testSocket);
+      EchoRandomData<8192>(testSocket);
+      // The websocket protocol treats lengths of >65536 specially.
+      EchoRandomData<65535>(testSocket);
+      EchoRandomData<65536>(testSocket);
+      EchoRandomData<131072>(testSocket);
```

Comment:
What is special about 131072.  Can you add a comment

Reply:
131072 is 2^17.